llvm/include/llvm/ADT/PointerUnion.h

```diff
@@ -259,6 +259,17 @@ struct CastInfo<To, const PointerUnion<PTs...>>
                                           CastInfo<To, PointerUnion<PTs...>>> {
 };
 
+// The default implementation of isPresent() for nullable types returns true
+// if the active member is not the first one, even if its value is nullptr.
+// Override the default behavior to return false for all possible null values.
+template <typename... PTs>
+struct ValueIsPresent<PointerUnion<PTs...>,
+                      std::enable_if_t<IsNullable<PointerUnion<PTs...>>>> {
+  using Union = PointerUnion<PTs...>;
+  static bool isPresent(const Union &V) { return static_cast<bool>(V); }
+  static decltype(auto) unwrapValue(Union &V) { return V; }
```

Comment:
```suggestion
  static Union &unwrapValue(Union &V) { return V; }
```

I don't think `decltype(auto)` helps here?